src/cpu/SearchBlock.zig

```diff
@@ -0,0 +1,344 @@
+            count += @intCast(add_count & 0xf);
+
+            var sub_count: u16 = 0;
+            inline for (sub) |location| sub_count += @as(u8, @bitCast(self.data[x * size + z + location.x * size + location.z]));
+            count -= @intCast(sub_count & 0xf);
+
+            if (count >= params.threshold) {
+                @setCold(true);
+                sufficiently_slimy_chunks += 1;
+                const real_x = @as(i32, @intCast(x + offset)) + self.min_x;
+                const real_z = @as(i32, @intCast(z + offset)) + self.min_z;
+
+                if (real_x >= params.x0 and real_x < params.x1 and
+                    real_z >= params.z0 and real_z < params.z1)
+                {
+                    resultCallback(context, .{
+                        .x = real_x,
+                        .z = real_z,
+                        .count = count,
+                    });
+                }
+            }
+        }
+    }
+    return sufficiently_slimy_chunks;
+}
+
+/// For every chunk within the searched area defined by this `SearchBlock`
+/// checks whether the amount of slime chunk slime chunks in spawn range of a player
+/// at the center of each chunk meets the given `threshold`
+pub fn calculateSliminessUncached(
+    self: *@This(),
+    comptime threshold: u8,
+    context: anytype,
+    comptime resultCallback: fn (@TypeOf(context), slimy.Result) void,
+) void {
+    for (0..size - mask.len) |x| {
+        for (0..size - mask.len) |z| {
+            var count: u8 = 0;
+            for (comptime getDistMask(), 0..) |row, i| {
+                for (row, 0..) |b, j| {
+                    count += @intFromBool(b and @import("slime_check/scalar.zig").isSlime(0x51133, @intCast(x + i), @intCast(z + j)));
+                }
+            }
+            if (count >= threshold) {
+                @setCold(true);
+                // @branchHint(.unlikely);
+
+                resultCallback(context, .{
+                    .x = @as(i32, @intCast(x + offset)) + self.min_x,
+                    .z = @as(i32, @intCast(z + offset)) + self.min_z,
+                    .count = count,
+                });
+            }
+        }
+    }
+}
+
+pub fn calculateSliminessForLocation(world_seed: i64, x: i32, z: i32) u8 {
+    var count: u8 = 0;
+    for (0..mask.len) |x_0| {
+        for (0..mask.len) |z_0| {
+            count += @intFromBool((comptime getDistMask())[x_0][z_0] and scalar.isSlime(world_seed, x + @as(i32, @intCast(x_0)) - offset, z + @as(i32, @intCast(z_0)) - offset));
+        }
+    }
+    return count;
+}
+
+pub fn format(self: @This(), comptime fmt: []const u8, options: std.fmt.FormatOptions, writer: anytype) !void {
+    const width, const height = comptime blk: {
+        if (fmt.len == 0) break :blk .{ 32, 32 };
+        if (fmt[0] != '[' or fmt[fmt.len - 1] != ']') @compileError("format for ProcessingChunk must be surrounded with brackets");
```

Comment:
brackets have special meaning to `std.fmt` - is there a reason you're passing in a size, rather than using the block's actual size?

Also, looks like you've got some left over stuff from a rename :)

Reply:
> is there a reason you're passing in a size, rather than using the block's actual size?

printing the entire block wraps around (at least on my terminal) so i just made it configurable

> Also, looks like you've got some left over stuff from a rename :)

? i'm not quite sure i follow